sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/IntermediateType.kt

```diff
@@ -16,135 +16,99 @@
 package app.cash.sqldelight.core.lang
 
 import app.cash.sqldelight.core.compiler.integration.adapterName
-import app.cash.sqldelight.core.dialect.api.DialectType
 import app.cash.sqldelight.core.lang.psi.ColumnTypeMixin
 import app.cash.sqldelight.core.lang.util.isArrayParameter
+import app.cash.sqldelight.dialect.api.IntermediateType
 import com.alecstrong.sql.psi.core.psi.Queryable
-import com.alecstrong.sql.psi.core.psi.SqlBindExpr
-import com.alecstrong.sql.psi.core.psi.SqlColumnDef
 import com.intellij.psi.util.PsiTreeUtil
 import com.squareup.kotlinpoet.CodeBlock
 import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
-import com.squareup.kotlinpoet.TypeName
 import com.squareup.kotlinpoet.asClassName
 
+internal fun IntermediateType.argumentType() = if (bindArg?.isArrayParameter() == true) {
```

Comment:
Everything in this file just moved from being a member function to an extension function